firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/FolderHierarchyFetcher.swift

```diff
@@ -35,22 +35,23 @@ struct DefaultFolderHierarchyFetcher: FolderHierarchyFetcher, BookmarksRefactorF
 
     func fetchFolders() async -> [Folder] {
         let numDesktopBookmarks = await countDesktopBookmarks()
-        return await withCheckedContinuation { continuation in
-            profile.places.getBookmarksTree(rootGUID: rootFolderGUID,
-                                            recursive: true).uponQueue(.main) { data in
-                var folders = [Folder]()
-                defer {
-                    continuation.resume(returning: folders)
-                }
-                guard let rootFolder = data.successValue as? BookmarkFolderData else { return }
-                let hasDesktopBookmarks = (numDesktopBookmarks ?? 0) > 0
-
-                let childrenFolders = rootFolder.children?.compactMap {
-                    return $0 as? BookmarkFolderData
-                }
+        return await withUnsafeContinuation { continuation in
+            profile.places.getBookmarksTree(rootGUID: rootFolderGUID, recursive: true) { result in
+                switch result {
+                case .success(let data):
+                    guard let rootFolder = data as? BookmarkFolderData else { return }
```

Comment:
Hi @PARAIPAN9 ! @lmarceau just wanted me to take a peek at the continuations.

There's a chance we miss calling continuation.resume here in the guard return since we don't have `defer` anymore. Are we explicitly removing `defer`s as well as the `Deferred`?